### PR TITLE
Update `Snap geometry edits` sample to include geometry guides

### PR DESCRIPTION
**Pull request by duffh**
## Description

PR to update Kotlin sample _"Sample geometry edits"_ in `Edit and Manage data` category.

## Links and Data
Sample Epic: [runtime/bistromath#2714](https://devtopia.esri.com/runtime/bistromath/issues/2714)
- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/sampleviewer/) Job for this PR has been run
  - [x] https://runtime-kotlin.esri.com/view/vTest/job/vtest/job/sampleviewer/33/
  - [x] https://runtime-kotlin.esri.com/view/all/job/vtest/job/sampleviewer/42/
  
## What To Review

-  Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

## How to Test

Run the sample on the sample viewer or the repo.

### CODE REVIEW

**Comment by fpearson1111** [comment]
@shubham7109 I can take care of the review comments for this while Hamish is out (he is back on Monday AFAIK)

**Comment by fpearson1111** [comment]
@shubham7109 Do you know what is causing the error in the latest vTest run about duplicate classes? I'm guessing some kind of dependency clash between the samples app and the Maps SDK?

File: snap-geometry-edits/README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ A field worker can create new features by editing and snapping the vertices of a
 
 To create a geometry, press the create button to choose the geometry type you want to create (i.e. points, multipoints, polyline, or polygon) and interactively tap and drag on the map view to create the geometry.
 
-To configure snapping, press the snap settings button to enable or disable snapping and choose which snap sources to snap to.
+Snap settings can be configured by enabling and disabling snapping, feature snapping, geometry guides and snap sources.
```
Comment (shellygill) [inline snap-geometry-edits/README.md:15]: FYI I believe it's standard to use the oxford comma in lists like this that end with a conjunction (i.e. use a comma before the "and"):

```suggestion
Snap settings can be configured by enabling and disabling snapping, feature snapping, geometry guides, and snap sources.
```

File: snap-geometry-edits/README.md [hunk 2]
```diff
@@ -34,7 +34,8 @@ To save your edits, press the save button.
 4. Create a `ReticleVertexTool` and set it into the `GeometryEditor.tool`.
 5. Call `syncSourceSettings` after the map's operational layers are loaded and the geometry editor connected to the map view.
 6. Set `SnapSettings.isEnabled` and `SnapSourceSettings.isEnabled` to true for the `SnapSource` of interest.
-7. Start the geometry editor with a `GeometryType`.
+7. Toggle geometry guides using `SnapSettings.isGeometryGuidesEnabled` and feature snapping using `SnapSettings.isFeatureSnappingEnabled`.
```
Comment (shellygill) [inline snap-geometry-edits/README.md:37]: I'd suggest changing "toggle" to something more definitive here, e.g. 
```suggestion
7. Enable or disable geometry guides using `SnapSettings.isGeometryGuidesEnabled` and feature snapping using `SnapSettings.isFeatureSnappingEnabled`.
```

File: snap-geometry-edits/README.md [hunk 3]
```diff
@@ -62,6 +63,8 @@ To snap to polygon and polyline layers, the recommended approach is to set the `
 
 Snapping can be used during interactive edits that move existing vertices using the `VertexTool` or `ReticleVertexTool`. It is also supported for adding new vertices for input devices with a hover event (such as a mouse move without a mouse button press). Using the `ReticleVertexTool` to add and move vertices allows users of touch screen devices to clearly see the visual cues for snapping.
 
+Geometry guides are enabled by default when snapping is enabled. These allow for snapping to a point coinciding with, parallel to, perpendicular to or extending an existing geometry.
```
Comment (shellygill) [inline snap-geometry-edits/README.md:66]:
```suggestion
Geometry guides are enabled by default when snapping is enabled. These allow for snapping to a point coinciding with, parallel to, perpendicular to, or extending an existing geometry.
```

**Review by shellygill — COMMENTED** [review]
A few v minor readme suggestions @fpearson1111 / @duffh.

**Comment by shubham7109** [comment]
@fpearson1111 vTests should now be stable. Created a vTest for this PR: [Link](https://runtime-kotlin.esri.com/view/all/job/vtest/job/sampleviewer/42/)

I see a crash everytime I try to open the bottom sheet when I click on the "Snap Settings" button without selecting a geometry type: 

````Kotlin
// In showBottomSheet():
geometryEditor.snapSettings.syncSourceSettings()

// Crashes with error:
Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x0 in tid 16193 (apgeometryedits), pid 16193 (apgeometryedits)
````

**Review by shellygill — APPROVED** [review]
Thanks @fpearson1111 for doc changes.

Good catch @shubham7109 - the sync issue is now fixed in core in the most recent build.

**Comment by fpearson1111** [comment]
@shubham7109 Ah I think that was a problem in recent builds of the Maps SDK, but it should be fixed now. I'll increment the build number.

**Review by shubham7109 — APPROVED** [review]
LGTM 👍

**Comment by fpearson1111** [comment]
Thanks @shellygill @shubham7109

**Pull request merged**